pytest_httpserver/httpserver.py

```diff
@@ -51,14 +53,44 @@ class RequestMatcher:
     :param query_string: the http query string starting with ``?``, such as ``?username=user``
     """
 
+    class HeaderValueMatcher:
+        """
+        Matcher object for the header value of incoming request.
+
+        :param matchers: mapping from header name to comparator function that accepts actual and expected header values
+            and return whether they are equal as bool.
+        """
+        DEFAULT_MATCHERS = {}
+
+        def __init__(self, matchers: Optional[Mapping[str, Callable[[str, str], bool]]] = None):
+            self.matchers = self.DEFAULT_MATCHERS if matchers is None else matchers
+
+        @staticmethod
+        def authorization_header_value_matcher(actual: str, expected: str) -> bool:
+            return parse_authorization_header(actual) == parse_authorization_header(expected)
+
+        @staticmethod
+        def default_header_value_matcher(actual: str, expected: str) -> bool:
+            return actual == expected
+
+        def __call__(self, header_name: str, actual: str, expected: str) -> bool:
+            matcher = self.matchers[header_name]
```

Comment:
what would happen if self.matchers[header_name] raises a KeyError? Do you want to handle it or are you ok with raising a KeyError here? Just asking...

Reply:
Yep, you are right, I reraise custom exception.